src/main/scala/scala/collection/decorators/IteratorDecorator.scala

```diff
@@ -72,4 +72,44 @@ class IteratorDecorator[A](val `this`: Iterator[A]) extends AnyVal {
     loop(immutable.List.empty)
   }
 
+  /**
+    * Constructs an iterator where consecutive elements are accumulated as
+    * long as the output of f for each element doesn't change.
+    * <pre>
+    * Vector(1,2,2,3,3,3,2,2)
+    *     .iterator
+    *     .groupUntilChanged(identity)
+    *     .toList
+    * </pre>
+    * produces
+    * <pre>
+    * List(Seq(1),
+    *      Seq(2,2),
+    *      Seq(3,3,3),
+    *      Seq(2,2))
+    * </pre>
+    * @param f the function to compute a key for an element
+    * @tparam K the type of the computed key
+    * @return an iterator of sequences of the consecutive elements with the
+    *         same key in the original iterator
+    */
+  def groupUntilChanged[K](f: A => K): Iterator[Iterable[A]] =
+    new AbstractIterator[Seq[A]] {
+      private var hd: Option[A] = `this`.nextOption()
+      override def hasNext: Boolean = hd.isDefined
+
+      override def next(): Seq[A] = {
+        hd match {
+          case None => Iterator.empty.next()
+          case Some(head) =>
+            hd = `this`.nextOption()
+            var seq = mutable.Buffer(head)
+            while (hd.exists(el => f(el) == f(head))) {
+              seq = seq ++ hd
+              hd = `this`.nextOption()
+            }
+            seq.toVector
```

Comment:
`seq = seq ++ hd` will create a clone of `mutable.Buffer(head)` and append the element to it. And then at the end `toVector` will have to copy again. It would be much more efficient to just create a `Vector.newBuilder[A]` and append each element to that.

Reply:
ack